src/IR.h

```diff
@@ -205,7 +205,8 @@ struct Load : public ExprNode<Load> {
     // If it's a load from an image parameter, this points to that
     Parameter param;
 
-    EXPORT static Expr make(Type type, std::string name, Expr index, Buffer<> image, Parameter param);
+    EXPORT static Expr make(Type type, std::string name, Expr index, Buffer<> image,
+                            Parameter param, Expr predicate);
```

Comment:
That's very dangerous - it would make it easy to silently drop a predicate in a mutator.